server/routerlicious/packages/lambdas/src/nexus/index.ts

```diff
@@ -1097,20 +1118,38 @@ export function configureWebSocketServices(
 						socket.emit("nack", "", [nackMessage]);
 						return;
 					}
-					contentBatches.forEach((contentBatch) => {
-						const contents = Array.isArray(contentBatch)
-							? contentBatch
-							: [contentBatch];
 
-						for (const content of contents) {
+					if (supportedFeaturesMap.get(clientId)?.[feature_submit_signals_v2]) {
+						for (const signal of contentBatches as ISentSignalMessage[]) {
```

Comment:
`as` is really quite dangerous and should be avoided in general.
We should not be assuming `contentBatches` is an `ISentSignalMessage[]` based on anything but `contentBatches` itself. The precondition with feature check is good, but not sufficient.
For each item we'll need to check that what it is; make sure it conforms to `ISentSignalMessage`.